kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/EvalHelper.java

```diff
@@ -38,8 +38,25 @@
 public class EvalHelper {
     public static final Logger LOG = LoggerFactory.getLogger( EvalHelper.class );
 
+    private static String removeBlankSpaces(String variableName) {
+        StringBuilder sb = new StringBuilder(variableName);
+        int j = 0;
+        boolean lastSpace = false;
+        for (int i = 0; i < sb.length(); i++) {
+            if (!Character.isWhitespace(sb.charAt(i))) {
+                sb.setCharAt(j++, sb.charAt(i));
+                lastSpace = false;
+            } else if (!lastSpace) {
+                sb.setCharAt(j++, sb.charAt(i));
+                lastSpace = true;
+            }
+        }
+        sb.delete(j, sb.length());
+        return sb.toString();
+    }
+
     public static String normalizeVariableName(String name) {
-        return name.replaceAll( "\\s+", " " );
```

Comment:
The regex-based implementation can be made noticeably faster if the pattern compilation step [replaceAll javadoc](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#replaceAll(java.lang.String,%20java.lang.String) is extracted e.g. as a static field of this helper.
I would prefer the following implementation:
```
public class EvalHelper {
    private static final Pattern SPACES_PATTERN =  Pattern.compile("\\s+");

   public static String normalizeVariableName(String name) {
     return SPACES_PATTERN.matcher(name).replaceAll("");
  }
}
```